emoji-substitution/substitute.js

```diff
@@ -0,0 +1,93 @@
+/*
+ * This file is responsible for performing the logic of replacing
+ * all occurrences of each mapped word with its emoji counterpart.
+ */
+
+// emojiMap.js defines the 'sortedEmojiMap' variable.
+// Referenced here to reduce confusion.
+const emojiMap = sortedEmojiMap;
+
+/*
+ * For efficiency, create a word --> search RegEx Map too.
+ */
+let regexs = new Map();
+for (let word of emojiMap.keys()) {
+  // We want a global, case-insensitive replacement.
+  // @see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp
+  regexs.set(word, new RegExp(word, 'gi'));
```

Comment:
Yeah, I went back and forth on this one.  From the PR description:

> The regular expression used matches on substrings. For example, man is replaced with 👨, and while the word man is replaced correctly throughout the document, so too are words like many (👨y), manipulate (👨ipulate), command (com👨d), etc.
I almost re-wrote this out, but it's actually kind of fun 😄